Lib/test/test_type_cache.py

```diff
@@ -0,0 +1,41 @@
+""" Tests for the internal type cache in CPython. """
+import unittest
+from collections import Counter
+from test import support
+try:
+    from sys import _clear_type_cache, _get_type_version_tag
+except ImportError:
+    _clear_type_cache = None
+
+msg = "requires sys._clear_type_cache and sys._get_type_version_tag"
+@unittest.skipIf(_clear_type_cache is None, msg)
+@support.cpython_only
+class TypeCacheTests(unittest.TestCase):
+    def test_tp_version_tag_unique(self):
+        """tp_version_tag should be unique assuming no overflow, even after
+        clearing type cache.
+        """
+        all_version_tags = []
+        append_result = all_version_tags.append
+        # Note: try to avoid any method lookups within this loop,
+        # It will affect global version tag.
+        for _ in range(30):
+            _clear_type_cache()
+            class C:
+                # __init__ is only here to force
+                # C.tp_version_tag to update when initializing
+                # object C().
+                def __init__(self):
+                    pass
+            x = C()
+            tp_version_tag_after = _get_type_version_tag(C)
+            # Try to avoid self.assertNotEqual to not overflow type cache.
+            assert tp_version_tag_after != 0
```

Comment:
You could use the same trick as above (`append_result = all_version_tags.append`) to use `TestCase.assertNotEqual` without the `LOAD_METHOD` here.

Reply:
``list.append`` only looks up ``list``, while ``self.assertNotEqual`` calls multiple other methods inside (possibly updating the tag multiple times). However, testing locally shows that it should still always trip the test. So I'll do that for readability.

BTW, I realised my code here didn't take into account if the version counter overflowed. So I'm amending it now (and moving the methods over to `_testcapi` instead of `sys`.